atoma-service/src/streamer.rs

```diff
@@ -563,8 +567,19 @@ impl Streamer {
 
         let (signature, response_hash) = self.sign_chunk(&chunk)?;
 
+        let intra_token_generation_timer = CHAT_COMPLETIONS_INTRA_TOKEN_GENERATION_TIME
+            .with_label_values(&[&self.model])
+            .start_timer();
+        self.intra_token_generation_timer = Some(intra_token_generation_timer);
+
         let choices = match chunk.get(CHOICES).and_then(|choices| choices.as_array()) {
-            Some(choices) => choices,
+            Some(choices) => {
+                // Observe the intra-token generation timer
+                if let Some(timer) = self.intra_token_generation_timer.take() {
```

Comment:
I would observe the timer at the end of the logical flow for the chunks (slightly down the line)

Reply:
Updated in 540c294

I think it's reasonable to assume this metric excludes the time taken to handle the encryption request for each chunk

Reply:
I would add it though, as we want to have detailed metrics on total time to generate next token, and if running with confidential compute mode on, encryption/decryption should be taken in consideration

Reply:
That's a valid point but I think we should perhaps track that separately to see if encryption requests are a potential bottleneck, token generation from from my perspective should be viewed as a separate process.

30fcf2f

Reply:
Sure, I can agree to that. I've removed the encryption related metrics and updated the PR 7b55718